--- a/python/plugins/processing_whitebox/whiteboxProvider.py
+++ b/python/plugins/processing_whitebox/whiteboxProvider.py
@@ -64,11 +64,11 @@
         ProcessingConfig.addSetting(Setting(self.name(),
                                             whiteboxUtils.WHITEBOX_ACTIVE,
                                             self.tr('Activate'),
-                                            False))
+                                            True))
         ProcessingConfig.addSetting(Setting(self.name(),
                                             whiteboxUtils.WHITEBOX_EXECUTABLE,
                                             self.tr('WhiteboxTools executable'),
-                                            whiteboxUtils.whiteboxToolsExecutable(),
+                                            "/usr/local/opt/whitebox-tools/bin/whitebox_tools",
                                             valuetype=Setting.FILE))
         ProcessingConfig.addSetting(Setting(self.name(),
                                             whiteboxUtils.WHITEBOX_VERBOSE,
